### PR TITLE
Fix a bug where spaces around link anchors were lost

### PR DESCRIPTION
**Pull request by hosamaly**
As an example, the following HTML:
```html
<p>Orbital play <a>Brighton Racecourse</a> on 29 June and <a>Zebedee’s Yard, Hull,</a> on 30 June. <a>Then touring</a> until 10 August.</p>
```
used to result in:
```html
<p>Orbital play BrightonRacecourseon 29 June andZebedee’s Yard, Hull, on 30 June.Then touringuntil 10 August.</p>
```
This was caused by the call to `Utility.trim`. We used to trim the input so that we won't wrap the spaces between XML tags (e.g. `<body> <body.content>`) in paragraphs. This commit fixes the logic that would have caused the incorrect wrapping in the first place.

### CODE REVIEW

**Comment by LATaylor-guardian** [comment]
👍

File: src/main/scala/com/gu/kpp/nitf/XhtmlToNitfTransformer.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ class XhtmlToNitfTransformer(config: HtmlToNitfConfig) {
   }
 
   // the order of these rules is important
-  private def transformationRules = Seq(
+  private lazy val transformationRules = Seq(
```
Comment (tomrf1) [inline src/main/scala/com/gu/kpp/nitf/XhtmlToNitfTransformer.scala]: why is this lazy?

Comment (tomrf1) [reply src/main/scala/com/gu/kpp/nitf/XhtmlToNitfTransformer.scala]: Let's avoid lazy unless it's essential

Comment (hosamaly) [reply src/main/scala/com/gu/kpp/nitf/XhtmlToNitfTransformer.scala]: Done

**Pull request merged**